### ISSUE TITLE
BHĀGAVATAPURĀṆA link target

### ISSUE THREAD

**Issue by funderburkjim** [sanskrit-lexicon/PWG#81]
Suggest we  make a link target for BHĀGAVATAPURĀṆA 
`18861 matches in 18837 lines for "<ls>BHĀG. P\. " in buffer: pwg.txt`

## pdf source suggestion
[archive org download](https://archive.org/download/bhagavata_purana_202001/bhagavata_purana.pdf)  looks of good quality.  
Produced by MAHARISHI UNIVERSITY OF MANAGEMENT VEDIC LITERATURE COLLECTION.

Individual pages are NOT images , as only 3.7MB for 1260 pages.  

A couple of random checks suggests that the numeration of स्कन्ध, अध्याय , श्लोक agrees with that  of PWG.
For example, under PWG विनष्टि headword is found `{#suhfdvinazwi#}
<ls>BHĀG. P. 3,1,21.</ls>`
and सुहृद्विनष्टिं is found on page 111.

## link target ui suggestion
Two possibilities seem relevant
* like Manu:  https://sanskrit-lexicon-scans.github.io/manu/index.html?8,43
* Like R. Gorr. : https://sanskrit-lexicon-scans.github.io/ramayanagorr/?2,27,7

I suspect either one could be used as model for Bhagavata.

## repository name  
A repository in sanskrit-lexicon-scans will contain the pdfs and user-interface.  Should the name of this repo be "bhagavata"  (name will appear in links, e.g. sanskrit-lexicon-scans.github.io/bhagavata/?1,2,3)

## repository name revised to bhagp_bur
Sample links:
https://sanskrit-lexicon-scans.github.io/bhagp_bur/app1/

https://sanskrit-lexicon-scans.github.io/bhagp_bur/app1/?2,4,6

https://sanskrit-lexicon-scans.github.io/bhagp_bur/app1/index.html?2,4,6

A link to the repository contents (for revisions, etc.) is
https://github.com/sanskrit-lexicon-scans/bhagp_bur

------------------------------------------
## Devanagari oddity
When one selects and copies some Devanagari text from the pdf, and then pastes into a text editor,  the pasted text is NOT Devanagari.  E.g. try 
copying स्कन्धः from page 109 of the pdf; when pasted into Emacs text one sees `SkN/"` .

**Comment by funderburkjim** [comment]
Three questions before I begin implementation:
1) archive org source ok?
2) which UI should be developed?
3) 'bhagavata' ok for repository name?

Request others to provide feedback.

**Comment by Andhrabharati** [comment]
Though there aren't many variants in the **BHĀG. P.** texts (in vogue), just like to mention that PWG has cited from Burnouf's ed. **as the primary source** (and also has some citations from its French Introduction) for the first 9 skandhas (~16k of them) and for the last 3 skandhas [10-12] (~4k of them) used the Bomb. ed. [Ganapat Krishnaji, Gujarati Press].

**Comment by funderburkjim** [comment]
### burnouf bhagavata notes

* volume 1 skandhas 1-3 (google books)
* volume 2 skandhas 4-6 (archive org -- the link mentioned above)
* volume 3 skandhas 7-9 (google books)
* volume 4 skandha 10, first part (archive.org)


* no source found for "Bomb. ed. [Ganapat Krishnaji, Gujarati Press"

-----------------------------

### Description of pwg and pw listings for "BHĀG. P." literary source:

```
PWG:
BHĀGAVATAPURĀṆA, nach Anführungen im VP. und ŚKDR. Von ऊ 
an sind die 9 von BURNOUF edirten Skandha (Gild. Bibl. 125) 
ausgebeutet worden.

Google translate:
According to quotes in VP. and ŚKDR. From ऊ 
onwards, the 9 Skandha (Gild. Bibl. 125) edited by BURNOUF have 
been exploited.

PW:
BHĀGAVATAPURĀṆA. Die 9 ersten Skandha nach BURNOUF's. 
Ausgabe, wenn nicht ausdrücklich. ed. Bomb. hinzugefügt wird; 
die letzten Skandha nach der ed. Bomb. In Klammern 
eingeschlossene Zahlen verweisen auf ed. Bomb.

Google translate:
The first 9 Skandha according to BURNOUF's. edition, unless 
expressly added. ed. Bomb.; the last Skandha according to the ed. 
Bomb. Numbers enclosed in parentheses refer to ed. Bomb.
```
------------------------------

**Comment by funderburkjim** [comment]
The Burnouf material mentioned above contains French translation and comments.  There is no Sanskrit text!  I think this disqualifies Burnouf for cdsl link target.

**Comment by funderburkjim** [comment]
I did further checking that  MUM version mentioned above.
For each skandha, I chose five or so PWG references from the first adhyaya,  and compared the corresponding verse in  MUM  (so about 60 comparisons altogether).

**All comparisons looked plausible.**

The MUM version looks to me like a good cdsl link target.

@Andhrabharati Do you see any problems with using this MUM version?

**Comment by Andhrabharati** [comment]
I've no issues if the MUM version is decided "to be used", because it is "easily available"; I was just pointing to what the "actual sources" used in PWG and pwk were.

Surprised to see this post--

> The Burnouf material mentioned above contains French translation and comments. There is no Sanskrit text! I think this disqualifies Burnouf for cdsl link target.

This clearly shows that Jim could not get to see the actual Burnouf edition volumes!!
[Indeed, it is somewhat difficult to reach out these volumes, while the French translation volumes could be easily "got".]

Here are the 3 volumes of Burnouf **with Sanskrit text**--

![image](https://github.com/user-attachments/assets/4ab1e8fe-94af-42f4-af72-870612d140ed)
![image](https://github.com/user-attachments/assets/15d1548f-4d3c-474d-9963-6cbc3955cff3)

![image](https://github.com/user-attachments/assets/6277ac7b-5203-449f-ada8-6531ba279dc2)
![image](https://github.com/user-attachments/assets/102a5aec-5eed-4671-a0db-b40c801ae279)

![image](https://github.com/user-attachments/assets/b3222e97-e742-4a2b-b48f-b9325047523f)
![image](https://github.com/user-attachments/assets/61ade629-b71b-4bf8-81e5-44a60cd2ac5e)

and against the post

> no source found for "Bomb. ed. [Ganapat Krishnaji, Gujarati Press"

these may be looked at--
[Vol.1] https://www.digitale-sammlungen.de/en/view/bsb10211024?page=7
[Vol.2] https://www.digitale-sammlungen.de/en/view/bsb10625867?page=5

**Comment by funderburkjim** [comment]
> in a way has questioned the very intent 

Not at all; I did not find a reference to the Devanagari edition, so did not know it exists.
Now your pictures prove the existence of the Devanagari edition!
Will you provide links for the Devanagari editions?

**Comment by Andhrabharati** [comment]
[Vol.1]
https://gallica.bnf.fr/ark:/12148/bpt6k97544594?rk=21459;2

[Vol.2]
https://gallica.bnf.fr/ark:/12148/bpt6k9753913j.r=Le%20Bh%C3%A2gavata%20Pur%C3%A2na?rk=42918;4

[Vol.3]
https://gallica.bnf.fr/ark:/12148/bpt6k9756588t.r=Le%20Bh%C3%A2gavata%20Pur%C3%A2na?rk=64378;0

**Comment by Andhrabharati** [comment]
Surprised to see Jim still awake & active, so late a night-time at his place!

**Comment by Andhrabharati** [comment]
And here is the index file for Burnouf's Bh.P.--
[Burnouf BhP index.txt](https://github.com/user-attachments/files/17840605/Burnouf.BhP.index.txt)

**Comment by Andhrabharati** [comment]
![image](https://github.com/user-attachments/assets/64b388d6-ad21-4645-9931-11ec53532343)

![image](https://github.com/user-attachments/assets/919a73bf-fb05-4e1f-a6a1-f6d74fe2d938)

![image](https://github.com/user-attachments/assets/6b8908c0-d6d4-4821-8249-6a7c75535ffe)

Yes, there seems to be some issue in downloading these now!

Granted permission.

**Comment by Andhrabharati** [comment]
These links are working at my end (as of now)--

https://gallica.bnf.fr/ark:/12148/bpt6k97534859.r=le%20bhagavata%20purana?rk=21459;2

https://gallica.bnf.fr/ark:/12148/bpt6k9759793r.r=le%20bhagavata%20purana?rk=42918;4

https://gallica.bnf.fr/ark:/12148/bpt6k9756588t.r=le%20bhagavata%20purana?rk=64378;0

**Comment by funderburkjim** [comment]
Got the files from Google Drive. All three contain the Devanagari text. Hurray!
So I can begin working with your index.

**Comment by funderburkjim** [comment]
I did try the [links above](https://github.com/sanskrit-lexicon/PWG/issues/81#issuecomment-2492351162),  and they all are as expected.

**Comment by funderburkjim** [comment]
As for the Bombay edition,  I have rotated the pages before extracting.
Agree?

We'll need an index also for Bombay edition.

My plan:  For each of Burnouf and Bombay
* make a repo in sanskrit-lexicon-scans
* make a simple UI (hosted at Github) like one of those in the first comment 
  * I'm leaning towards the Gorresio UI variant, which would be better as stand-alone 'reader'.  Any thoughts on this?
 
Then the display logic at cdsl would choose (at least for PWG) the burnouf display for skandhas 1-9, and the Bombay display for skandhas 10-12.

**Comment by Andhrabharati** [comment]
> We'll need an index also for Bombay edition.

Yes, and for the M.Bh. and R. as well.

**Comment by funderburkjim** [comment]
> Yes, and for the M.Bh. (scan links posted at github by AB) and R. (scan links not yet posted!) as well.

When this Bhagavata link target is finished, please remind me of what you are referring to here.

BTW:  I am also investigating a bugbounty security issue.

**Comment by Andhrabharati** [comment]
You have nothing to do about this, Jim!

I was just mentioning that those two also need to be indexed.

**Comment by funderburkjim** [comment]
Github app initialized.

Sample urls:
https://sanskrit-lexicon-scans.github.io/bhagp_bur/app1/

https://sanskrit-lexicon-scans.github.io/bhagp_bur/app1/?2,4,6

https://sanskrit-lexicon-scans.github.io/bhagp_bur/app1/index.html?2,4,6

Is this beautiful? Any warts to remove ?

Next step is to generate links for PWG et. al.

**Comment by drdhaval2785** [comment]
Looks beautiful to me.

**Comment by gasyoun** [comment]
> Is this beautiful? Any warts to remove ?

Hardly usable because of the scrolling.

![cxzcxzcxzczx](https://github.com/user-attachments/assets/5c0a18c5-5f79-4cf9-8414-949f7dcf71b3)

**Comment by funderburkjim** [comment]
@gasyoun Yes, I noticed that also.  How to remove 'scrolling' wart ?  I suspect this is an html-css issue.
Do you have anyone who can readily fix ?

**Comment by funderburkjim** [comment]
## link standardization for pwg
5000+ lines were changed in pwg, to improve markup for 'Bhāg. P.' links.
The changes can be seen in files change_0a.txt, change_1.txt and change_2.txt in [issue81 directory](https://github.com/sanskrit-lexicon/PWG/tree/master/pwgissues/issue81).  

A comparison of the links with those of the index for the Burnouf version of skandhas 1-9 was made, and uncovered a few errors; most of which were solved; See readme_check_bur.txt for the identification of these.

The work started with 22156 references, and now there are 30564.

Some summaries might be of interest.
* (30461) bhagp_standard_2.txt has all the standard 'Bhāg. P.' records 
* (103) bhagp_nonstandard_2.txt has other literary source items mentioning 'Bhāg. P.'. 
* bhagp_verse.txt has all the verses from the standard records, with their frequency.  The (0,0,0) item is `<ls>Bhāg. P.</ls>` . Altogether, 10509 different verses were referenced in PWG.

**Comment by Andhrabharati** [comment]
Here are the resolutions for the 20 "Unsolved Problems" (mentioned in readme_check_bur.txt)--
[readme_check_bur--- Unsolved Problems.txt](https://github.com/user-attachments/files/18038387/readme_check_bur---.Unsolved.Problems.txt)

[Jim needs to mark the print_changes, where required in these places, accordingly]

**Comment by funderburkjim** [comment]
@Andhrabharati  please check the revision
```
106569	sarj	995565	<ls>BHĀG. P. 4,31,32.</ls>  32 not found	
AB: it is at 4,11,32
```
At [4,11,32](https://sanskrit-lexicon-scans.github.io/bhagp_bur/app1/?4,11,32) there is `udvijate`,  but entry is about 'sarj'. [page 7-0802 udvi](https://www.sanskrit-lexicon.uni-koeln.de/scans/PWGScan/PWGScanpdf/pwg7-0801.pdf) 
I do see  उपसृष्ट  (which is from सर्ज् ) at 4,11,32 but was expecting to see 
something like उद्विसृष्ट .

**Comment by funderburkjim** [comment]
## implement the 'unsolved problems'.

Note: 4,11,32 was implemented, also. 

This finishes the work for standardizing the PWG links for BHĀGAVATAPURĀṆA
Will begin the same for PW(K)  (and then SCH and MW).

**Comment by Andhrabharati** [comment]
> @Andhrabharati please check the revision
> 
> ```
> 106569	sarj	995565	<ls>BHĀG. P. 4,31,32.</ls>  32 not found	
> AB: it is at 4,11,32
> ```

My error, @funderburkjim !

The citation refers to 4,31,22, and not to 4,11,32.

**Comment by funderburkjim** [comment]
## revision made.

The print changes to pwg uncovered in this issue were surprising to me.
I wonder if there are any additional 'automated' ways to validate BHĀGAVATAPURĀṆA references. Seems like a difficult NLP problem.

**Comment by funderburkjim** [comment]
For standardization of PW(K) references, see #109.

**Comment by funderburkjim** [comment]
@Andhrabharati   Have you prepared an index for Bombay edition?

**Comment by Andhrabharati** [comment]
No, I haven't touched it at all.

**Comment by funderburkjim** [comment]
Would making the Bombay index be a good task to ask Anna if she would do?

**Comment by Andhrabharati** [comment]
I haven't seen Anna refusing to take up any task so far!

**But, I presume it is a task far below her capacity!!**

**Comment by funderburkjim** [comment]
## finishing the links to bur/bom editions.

As mentioned in [the comment above](https://github.com/sanskrit-lexicon/PWG/issues/81#issuecomment-2487767917),  a link to bhagavata purana for
pwg for skandhas 1-9 should use Burnouf, and for skandhas 10-12 should use Bombay edition.

The cdsl displays now make this choice (see commit to csl-websanlexicon repo above).

I think this issue can now be closed [but note that in testing the links today, I found a few problematic -- I'll mention these in another issue].
